nodejs/extensions.bzl

```diff
@@ -1,40 +1,50 @@
 "extensions for bzlmod"
 
-load(":repositories.bzl", "DEFAULT_NODE_REPOSITORY", "DEFAULT_NODE_VERSION", "nodejs_register_toolchains")
+load(
+    ":repositories.bzl",
+    "DEFAULT_NODE_REPOSITORY",
+    "DEFAULT_NODE_URL",
+    "DEFAULT_NODE_VERSION",
+    "nodejs_register_toolchains",
+)
 
 def _toolchain_extension(module_ctx):
     registrations = {}
     for mod in module_ctx.modules:
         for toolchain in mod.tags.toolchain:
             if toolchain.name != DEFAULT_NODE_REPOSITORY and not mod.is_root:
                 fail("Only the root module may provide a name for the node toolchain.")
-
             if toolchain.name in registrations.keys():
                 if toolchain.name == DEFAULT_NODE_REPOSITORY:
                     # Prioritize the root-most registration of the default node toolchain version and
                     # ignore any further registrations (modules are processed breadth-first)
                     continue
-                if toolchain.node_version == registrations[toolchain.name].node_version and toolchain.node_version_from_nvmrc == registrations[toolchain.name].node_version_from_nvmrc:
+                if (toolchain.node_version != registrations[toolchain.name].node_version or
+                    toolchain.node_version_from_nvmrc != registrations[toolchain.name].node_version_from_nvmrc):
+                    fail("Multiple conflicting toolchains declared for name {} ({} and {})".format(
+                        toolchain.name,
+                        toolchain.node_version,
+                        registrations[toolchain.name].node_version,
+                    ))
+                elif toolchain.node_urls != registrations[toolchain.name].node_urls:
+                    fail("Multiple toolchains with conflicting urls declared for name {} ({} and {})".format(
+                        toolchain.name,
+                        toolchain.node_urls,
+                        registrations[toolchain.name].node_urls,
+                    ))
+                else:
                     # No problem to register a matching toolchain twice
                     continue
-                fail("Multiple conflicting toolchains declared for name {} ({} and {})".format(
-                    toolchain.name,
-                    toolchain.node_version,
-                    registrations[toolchain.name],
-                ))
             else:
-                registrations[toolchain.name] = struct(
-                    node_version = toolchain.node_version,
-                    node_version_from_nvmrc = toolchain.node_version_from_nvmrc,
-                    include_headers = toolchain.include_headers,
-                )
+                registrations[toolchain.name] = toolchain
 
-    for k, v in registrations.items():
+    for k, t in registrations.items():
```

Comment:
Any reason for this? Revert if not...

Reply:
it's now a `toolchain` rather than a `version`, but I can revert it.

Reply:
Done!